lib/buffer.js

```diff
@@ -1049,6 +1049,24 @@ function _fill(buf, value, offset, end, encoding) {
   return buf;
 }
 
+Buffer.prototype.toInt = function toInt(radix = 10) {
+  let data = 0;
+
+  for (let i = this.length - 1; i >= 0; i--) {
+    const value = this[i];
+
+    if (value < 48) {
+      throw new ERR_OUT_OF_RANGE('this[i]', '>=', 48);
+    } else if (value > 57) {
+      throw new ERR_OUT_OF_RANGE('this[i]', '<=', 57);
+    }
+
+    data += radix ** (this.length - 1 - i) * (value - 48);
```

Comment:
It seems the length of buffer is unbounded. `data` may exceed `Number.MAX_SAFE_INTERGER`. What should we do about it?